### PR TITLE
fix(developer): handle multiple instances of Server starting

### LINKED ISSUE #11261: bug(developer): Keyman Developer Server can fail with `Error: listen EADDRINUSE: address already in use :::8008`
Sentry Issue: [KEYMAN-DEVELOPER-1GF](https://keyman.sentry.io/issues/4637652951/?referrer=github_integration)

```
Error: listen EADDRINUSE: address already in use :::8008
  File "C:/Program%20Files%20(x86)/Keyman/Keyman%20Developer/server/build/src/index.js", line 68, in run
  File "C:/Program%20Files%20(x86)/Keyman/Keyman%20Developer/server/build/src/index.js", line 20, in <anonymous>
...
(1 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #11261.

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/server/src/index.ts [hunk 1]
```diff
@@ -130,4 +126,68 @@ export async function run() {
...
+      fs.closeSync(lockFileDescriptor);
+      lockFileDescriptor = null;
+      try {
+        fs.unlinkSync(lockFilename);
+      } catch(err) {
+        // This can happen if an unlink call is made on a file with an open
+        // descriptor, such as happens above; when the file is closed it is
+        // automatically deleted. On Windows at least.
+        //
+        // console.debug(`Did not unlink ${lockFilename}, probably already deleted`);
```
Comment (darcywong00) [inline developer/src/server/src/index.ts:188]: Does this need to return false?

Comment (mcdurdin) [reply developer/src/server/src/index.ts:188]: no, this `process.on('exit')` closure doesn't return any value.

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.312-beta](https://keyman.com/downloads/releases/beta/17.0.312)

**Pull request merged**